File: 35-search-insert-position/35-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if target>nums[-1]:
            return len(nums)
        elif target<nums[0]:
            return 0
        start, end = 0, len(nums)-1
        while start<end-1:
            mid = (start+end)//2
            if nums[mid]<target:
                start = mid
            elif nums[mid]>target:
                end = mid
            else:
                return mid
        return start+1 if nums[start]<target else start

File: 35-search-insert-position/35-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        if(target<nums[0])
            return 0;
        if(target>nums[nums.length-1])
            return nums.length;
        int start = 0, end = nums.length-1, mid;
        while(start<end){
            mid = (start+end)/2;
            if(nums[mid]<target)
                start = mid+1;
            else if(nums[mid]>target)
                end = mid-1;
            else
                return mid;
        }
        if(nums[start]<target)
            return start+1;
        else
            return start;
    }
}